v2/googlecloud-to-elasticsearch/src/main/java/com/google/cloud/teleport/v2/elasticsearch/templates/SpannerChangeStreamsToElasticsearch.java

```diff
@@ -0,0 +1,176 @@
+            ? Timestamp.MAX_VALUE
+            : Timestamp.parseTimestamp(options.getEndTimestamp());
+
+    // Add use_runner_v2 to the experiments option, since Change Streams connector is only supported
+    // on Dataflow runner v2.
+    List<String> experiments = options.getExperiments();
+    if (experiments == null) {
+      experiments = new ArrayList<>();
+    }
+    boolean hasUseRunnerV2 = false;
+    for (String experiment : experiments) {
+      if (experiment.toLowerCase().equals(USE_RUNNER_V2_EXPERIMENT)) {
+        hasUseRunnerV2 = true;
+        break;
+      }
+    }
+    if (!hasUseRunnerV2) {
+      experiments.add(USE_RUNNER_V2_EXPERIMENT);
+    }
+    options.setExperiments(experiments);
+
+    String metadataTableName =
+        options.getSpannerMetadataTableName() == null
+            ? null
+            : options.getSpannerMetadataTableName();
+
+    final RpcPriority rpcPriority = options.getRpcPriority();
+    pipeline
+        .apply(
+            SpannerIO.readChangeStream()
+                .withSpannerConfig(
+                    SpannerConfig.create()
+                        .withHost(ValueProvider.StaticValueProvider.of(options.getSpannerHost()))
+                        .withProjectId(projectId)
+                        .withInstanceId(instanceId)
+                        .withDatabaseId(databaseId))
+                .withMetadataInstance(metadataInstanceId)
+                .withMetadataDatabase(metadataDatabaseId)
+                .withChangeStreamName(changeStreamName)
+                .withInclusiveStartAt(startTimestamp)
+                .withInclusiveEndAt(endTimestamp)
+                .withRpcPriority(rpcPriority)
+                .withMetadataTable(metadataTableName))
+        .apply("DataChangeRecord To JSON", ParDo.of(new DataChangeRecordToJsonFn()))
+        .apply(
+            "WriteToElasticsearch",
+            WriteToElasticsearch.newBuilder()
+                .setOptions(options.as(SpannerChangeStreamsToElasticsearchOptions.class))
+                .build());
+
+    return pipeline.run();
+  }
+
+  static class DataChangeRecordToJsonFn extends DoFn<DataChangeRecord, String> {
+
+    @ProcessElement
+    public void process(@Element DataChangeRecord element, OutputReceiver<String> output) {
+      List<ColumnType> cols = element.getRowType();
+      ModType modType = element.getModType();
+      String tableName = element.getTableName();
+      element
+          .getMods()
+          .forEach(
+              mod -> {
+                JSONObject keysJsonExtended = new JSONObject(mod.getKeysJson());
+                keysJsonExtended.put("TableName", tableName);
+                if (modType == ModType.INSERT || modType == ModType.UPDATE) {
+                  JSONObject newValuesJson = new JSONObject(mod.getNewValuesJson());
+                  // add any properties that are present in newValuesJson to the set of keys to make
+                  // a partial (or complete) row reflecting
```

Comment:
nit: unnecessary line break.

Reply:
I assumed you meant the comment and tried to resolve by combining.. was that what you meant?